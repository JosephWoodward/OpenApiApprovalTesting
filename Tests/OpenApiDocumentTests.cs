using System.Threading.Tasks;
using Shouldly;
using Xunit;

namespace Tests
{
    public class OpenApiDocumentTests : IClassFixture<ApiFixture>
    {
        private readonly ApiFixture _apiFixture;

        public OpenApiDocumentTests(ApiFixture apiFixture)
        {
            _apiFixture = apiFixture;
        }

        [Fact]
        public async Task OpenApiDocumentHasChanged()
        {
            var client = _apiFixture.Create();

            var request = await client.GetAsync("/swagger/v1/swagger.json");
            var content = await request.Content.ReadAsStringAsync();

            content.ShouldMatchApproved();
        }
    }
}