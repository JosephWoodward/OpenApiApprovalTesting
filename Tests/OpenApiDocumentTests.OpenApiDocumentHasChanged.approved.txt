{
  "openapi": "3.0.1",
  "info": {
    "title": "My API",
    "version": "v1"
  },
  "paths": {
    "/api/Product": {
      "get": {
        "tags": [
          "Product"
        ],
        "responses": {
          "200": {
            "description": "Success",
            "content": {
              "text/plain": {
                "schema": {
                  "type": "array",
                  "items": {
                    "$ref": "#/components/schemas/ProductViewModel"
                  }
                }
              },
              "application/json": {
                "schema": {
                  "type": "array",
                  "items": {
                    "$ref": "#/components/schemas/ProductViewModel"
                  }
                }
              },
              "text/json": {
                "schema": {
                  "type": "array",
                  "items": {
                    "$ref": "#/components/schemas/ProductViewModel"
                  }
                }
              }
            }
          }
        }
      }
    }
  },
  "components": {
    "schemas": {
      "ProductViewModel": {
        "type": "object",
        "properties": {
          "productName": {
            "type": "string"
          },
          "id": {
            "type": "integer",
            "format": "int32"
          }
        },
        "additionalProperties": false
      }
    }
  }
}